lib/ecto/migration/runner.ex

```diff
@@ -264,6 +266,19 @@ defmodule Ecto.Migration.Runner do
 
   ## Helpers
 
+  defp perform_operation(module, operation) do
+    if module.__migration__[:disable_ddl_transaction] do
+      apply(module, operation, [])
+    else
+      try do
+        module.after_begin()
+        apply(module, operation, [])
+      after
+        module.before_commit()
```

Comment:
We don't need a try/catch here because `before_commit` should only run if we are going to commit... and we are not going to commit if there is an error. :)